### DISCUSSION TITLE
Hi, I have a question about inpaint resolution selection

### DISCUSSION THREAD

**Discussion by BNP1111** [lllyasviel/Fooocus#2695] [Q&A] [answered]
Hi, I have a question about inpaint resolution selection,
and I noticed that whatever resolution I choose, fooocus seems to automatically select the resolution based on the shape of my mask.
Even if I force it with the“Forced Overwrite of Generating Width” and“Forced Overwrite of Generating Height” settings in debug tools,
is there any way to use my own custom resolution? Even if I have to modify the code to try it out, thank you

**Comment by mashb1t** [comment]
you'll have to adjust the mask, either before uploading or by shaving it in python. See async worker for inpaint mask processing.

**Reply by xhoxye** [reply]
I have an idea that for the mask area less than 1024, can I first zoom in, then redraw, and then zoom out and fill it back, which is helpful for detail optimization

**Reply by BNP1111** [reply]
Thank you for your valuable time to explain,  But it should be my problem, I didn't describe it clearly.

I actually want to ask is: 
1、first I import a picture to fooocus inpaint, then I mask part of the Area，like:
![1](https://github.com/lllyasviel/Fooocus/assets/125350921/ccd0b06c-2c9e-4935-93b8-7fe016f9bc62)

2、I set the resolution I want（896✖1152）, and then click generate 
![2](https://github.com/lllyasviel/Fooocus/assets/125350921/d8ed6ba0-3c44-4c50-bdda-2fa71462a6e5)

3、in CMD, i found that fooocus automatically adjusted the resolution to 1024✖1024（I understand that this is an automatic calculation and not an error）
![3](https://github.com/lllyasviel/Fooocus/assets/125350921/d1fa39c3-0b89-4067-84b9-b988900baa2b)

4、and then I went on to force the width and length values here.
![4](https://github.com/lllyasviel/Fooocus/assets/125350921/54a6deba-b23c-49e2-be9f-d816d0c7e6b7)

5、Click generate and found that fooocus still automatically adjusted the resolution to 1024✖1024
![5](https://github.com/lllyasviel/Fooocus/assets/125350921/446ac90b-0371-4fa0-a370-a7f74111070a)


what i want is this resolution can be defined by me, such as: 1280✖1280 etc. (just to test the model) , if the system can not be set, even if temporarily let me know where to change the code, i do not know whether this wish can be realized, thank you again!

**Reply by xhoxye** [reply]
The mask's redraw size is not associated with the resolution option, and you can't affect the redraw size by setting the resolution

**Reply by BNP1111** [reply]
like change the code?

**Reply by mashb1t** [reply, marked as answer]
@BNP1111 this doesn't make sense. The resolution of the uploaded image + mask + output have to be the same, if not the mask will get deformed and cut off and the AI has to dynamically do outpainting as well as cut off, which isn't the expected behavior, especially as outpainting is only available in default inpainting mode as it requires a completely different set of parameters in order to do so.
=> won't do
